custom/enikshay/case_utils.py

```diff
@@ -103,32 +102,38 @@ def get_person_case_from_episode(domain, episode_case_id):
     )
 
 
-def get_open_occurrence_case_from_person(domain, person_case_id):
+def get_open_occurrence_case_from_person(domain, person_case_id, last_closed=False):
     """
     Gets the first open 'occurrence' case for the person
 
     Assumes the following case structure:
     Person <--ext-- Occurrence
-
+    last_closed would give the latest closed occurrence case if no open occurrence case found
     """
     case_accessor = CaseAccessors(domain)
     occurrence_cases = case_accessor.get_reverse_indexed_cases([person_case_id])
     open_occurrence_cases = [case for case in occurrence_cases
                              if not case.closed and case.type == CASE_TYPE_OCCURRENCE]
     if not open_occurrence_cases:
+        if last_closed:
+            all_occurrence_cases = [case for case in occurrence_cases if case.type == CASE_TYPE_OCCURRENCE]
```

Comment:
do you think you could just include this in the first loop conditionally? That way you don't have to loop through them twice.
```py
[case for case in occurrence_cases
if (last_closed or not case.closed) and case.type == CASE_TYPE_OCCURRENCE]
```

Reply:
Agree could be written better. Its just that we want to check for last closed only when there are no open cases so that change won't work. One way would be to fetch all cases, and then first check for any open case and if not then return the first closed case. That might be a little complex to read.

Reply:
oh right, that makes sense then.